mne/viz/utils.py

```diff
@@ -2242,3 +2250,192 @@ def _grad_pair_pick_and_name(info, picks):
     picks = list(sorted(set(picked_chans)))
     ch_names = [info["ch_names"][pick] for pick in picks]
     return picks, ch_names
+
+
+def _setup_plot_projector(info, noise_cov, proj=True, use_noise_cov=True,
+                          nave=1):
+    from ..cov import compute_whitener
+    projector = np.eye(len(info['ch_names']))
+    whitened_ch_names = []
+    if noise_cov is not None and use_noise_cov:
+        # any channels in noise_cov['bads'] but not in info['bads'] get
+        # set to nan, which means that they are not plotted.
+        # XXX This picking call mimics what is done in compute_whitener...
+        # should probably allow for other channel types, too (e.g., ECoG)!
+        data_picks = pick_types(info, meg=True, eeg=True, ref_meg=False,
+                                exclude=())
+        data_names = set(info['ch_names'][pick] for pick in data_picks)
+        # these can be toggled by the user
+        bad_names = set(info['bads'])
+        # these can't in standard pipelines be enabled (we always take the
+        # union), so pretend they're not in cov at all
+        cov_names = ((set(noise_cov['names']) & set(info['ch_names'])) -
+                     set(noise_cov['bads']))
+        # Actually compute the whitener only using the difference
+        whiten_names = cov_names - bad_names
+        whiten_picks = pick_channels(info['ch_names'], whiten_names)
+        whiten_info = pick_info(info, whiten_picks)
+        rank = _triage_rank_sss(whiten_info, [noise_cov])[1][0]
+        whitener, whitened_ch_names = compute_whitener(
+            noise_cov, whiten_info, rank=rank, verbose=False)
+        whitener *= np.sqrt(nave)  # proper scaling for Evoked data
+        assert set(whitened_ch_names) == whiten_names
+        projector[whiten_picks, whiten_picks[:, np.newaxis]] = whitener
+        # Now we need to change the set of "whitened" channels to include
+        # all data channel names so that they are properly italicized.
+        whitened_ch_names = data_names
+        # We would need to set "bad_picks" to identity to show the traces
+        # (but in gray), but here we don't need to because "projector"
+        # starts out as identity. So all that is left to do is take any
+        # *good* data channels that are not in the noise cov to be NaN
+        nan_names = data_names - (bad_names | cov_names)
```

Comment:
maybe the case for using set methods for better readability.

Reply:
I think set operations are clearly described using their arithmetic equivalents ...?